### ISSUE TITLE
Dry deploy property still attempts to connect to rio

### ISSUE THREAD

**Issue by hazeltorek** [wpilibsuite/deploy-utils#8]
When using the command `./gradlew deploy -Pdeploy-dry`, the deploy process still attempts to discover and connect to a roboRIO. 

This was tested with a Java command-based template and a C++ timed robot template and I still encountered the same issue both times. The attached screenshot is from the C++ test.

![image](https://user-images.githubusercontent.com/51520860/148828346-7653f845-8971-4662-b6db-30b184aac9f9.png)

**Comment by ThadHouse** [comment]
That's not surprising. I think I completely removed the dry deploy. I'll have to check. If I didn't I'll figure out what it needs to work.

**Comment by sciencewhiz** [comment]
What's your use case for a dry deploy?

**Comment by hazeltorek** [comment]
I use it when testing changes to `build.gradle` related to deploy where the changes can be observed locally. For example, I used it a lot when writing [this article](https://github.com/wpilibsuite/frc-docs/pull/1533) on FRC docs and making [this fix](https://github.com/wpilibsuite/frc-docs/pull/1641) to the same article.

If the use case is not significant enough for it to be implemented again that's fine, I just figured it was notable that it didn't work since it is mentioned in the readme.
